plugins/woocommerce/templates/checkout/thankyou.php

```diff
@@ -41,43 +41,70 @@
 
 			<p class="woocommerce-notice woocommerce-notice--success woocommerce-thankyou-order-received"><?php echo apply_filters( 'woocommerce_thankyou_order_received_text', esc_html__( 'Thank you. Your order has been received.', 'woocommerce' ), $order ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?></p>
 
-			<ul class="woocommerce-order-overview woocommerce-thankyou-order-details order_details">
+			<?php if ( $needs_verification ) : ?>
```

Comment:
(3/3) However, we still test it here (and, depending on the version of PHP in use, I think may result in a notice level error or higher being emitted).